File: space-game/Core.Tests/Classes/TextureTests.vb
Imports Core.Classes
Imports Core.Tests.Data
Imports NUnit.Framework

Namespace Classes
    Public Class TextureTests

        <TestCaseSource(GetType(TextureTestsData), NameOf(TextureTestsData.Equals_WhenComparing_ReturnsExpected_Data))>
        Public Sub Equals_WhenComparing_ReturnsExpected(this As Texture, that As Texture, expected As Boolean)
            Dim result = this.Equals(that)

            Assert.AreEqual(expected, result)
        End Sub
    End Class
End Namespace

File: space-game/Core.Tests/Classes/TextureTests.cs
using Core.Classes;
using Core.Tests.Data;
using NUnit.Framework;

namespace Core.Tests.Classes
{
    public class TextureTests
    {
        [TestCaseSource(typeof(TextureTestsData), nameof(TextureTestsData.Equals_WhenComparing_ReturnsExpected_Data))]
        public void Equals_WhenComparing_ReturnsExpected(Texture @this, Texture that, bool expected)
        {
            bool result = @this.Equals(that);

            Assert.AreEqual(expected, result);
        }
    }
}
